src/SearchField/index.jsx

```diff
@@ -0,0 +1,266 @@
+    const { focused } = this.state;
+    const { screenReaderText } = this.props;
+
+    return (
+      <Icon
+        className={[classNames(
+          FontAwesomeStyles.fa,
+          FontAwesomeStyles['fa-search'],
+          {
+            [styles['text-light']]: !focused,
+            [styles['text-primary']]: focused,
+          },
+        )]}
+        id={newId('icon-SearchField')}
+        screenReaderText={screenReaderText}
+      />
+    );
+  }
+
+  getClearButton() {
+    const { focused } = this.state;
+    const shouldShowClearButton = this.shouldShowClearButton();
+
+    return (
+      <Button
+        className={[classNames(
+          styles['border-0'],
+          styles['ml-1'],
+          styles['search-field-clear-btn'],
+        )]}
+        label={(
+          <small>
+            <Icon
+              className={[classNames(
+                FontAwesomeStyles.fa,
+                FontAwesomeStyles['fa-times'],
+                {
+                  [styles['text-light']]: !focused,
+                  [styles['text-muted']]: focused,
+                },
+              )]}
+              id={newId('icon-SearchField')}
+              screenReaderText="clear search"
+            />
+          </small>
+        )}
+        onClick={this.handleClear}
+        aria-hidden={!shouldShowClearButton}
+      />
+    );
+  }
+
+  getSearchButton() {
+    const { focused } = this.state;
+    const { hasClearButton, searchButtonLabel } = this.props;
+    const inputTextHasValue = this.inputTextHasValue();
+
+    return (
+      <Button
+        className={[classNames(
+          styles['search-field-search-btn-transition'],
+          {
+            [styles['ml-1']]: !hasClearButton,
+            [styles['text-muted']]: !focused && inputTextHasValue,
+            [styles['border-left']]: !focused && inputTextHasValue,
+            [styles['btn-outline-primary']]: focused && inputTextHasValue,
+          },
+        )]}
+        label={searchButtonLabel}
+        disabled={!this.inputTextHasValue()}
+        inputRef={(input) => { this.searchButton = input; }}
+        onClick={this.handleSubmit}
+      />
+    );
+  }
+
+  getSearchActionButtons() {
+    const buttons = [
+      this.getSearchButton(),
+    ];
+
+    if (this.shouldShowClearButton()) {
+      buttons.unshift(this.getClearButton());
+    }
+
+    return buttons;
+  }
+
+  shouldShowClearButton() {
+    return this.props.hasClearButton && this.inputTextHasValue();
+  }
+
+  inputTextHasValue() {
+    const { value } = this.state;
+    return value && value.trim().length > 0;
+  }
+
+  handleFocus(event) {
+    this.setState({ focused: true });
+    this.props.onFocus(event);
+  }
+
+  handleBlur(event) {
+    this.setState({ focused: false });
+    this.props.onBlur(event);
+  }
+
+  handleChange(value) {
+    this.setState({ value });
+    this.props.onChange(value);
+  }
+
+  handleClear() {
+    this.setState({ value: '' });
+    this.props.onChange('');
```

Comment:
Should there be a separate `onClear` prop? I could kind've see some use cases where it might be useful.

Reply:
Hmm, I could see how it'd be useful. However, the end user of this component could watch for clearing (i.e., value becomes an empty string) by using the existing `onChange` callback prop that's called in `handleClear`, so I'm not sure if it's worth adding here.